pkg/gitproviders/provider_org.go

```diff
@@ -125,3 +125,26 @@ func (p orgGitProvider) GetCommits(ctx context.Context, repoUrl RepoURL, targetB
 func (p orgGitProvider) GetProviderDomain() string {
 	return getProviderDomain(p.provider.ProviderID())
 }
+
+// GetRepoFiles returns the files found in a directory. The targetPath must point to a directory, not a file.
+// Note that the current implementation only gets an end subdirectory. It does not get multiple directories recursively. See https://github.com/fluxcd/go-git-providers/issues/143.
+func (p orgGitProvider) GetRepoFiles(ctx context.Context, repoUrl RepoURL, targetPath, targetBranch string) ([]*gitprovider.CommitFile, error) {
```

Comment:
The change in naming is subtle, but I don't really like it because this, in fact, does not just `get files from a directory`. It fetches things from a URL and then checks for content inside that. It does more and I think the name reflects that nicely that it gets files from a repository. It could be more bespoke, but I think GetFilesInDir would simplify that too much.

Reply:
Agree with Gergely, also this leaves it open to make use of Github's `RepositoryContentGetOptions` to get a repo's entire filesystem recursively. 

I opened an issue about this in go-git-providers: https://github.com/fluxcd/go-git-providers/issues/143